### ISSUE TITLE
Difficulties with Multi-line headers. Rows shifted down.

### ISSUE THREAD

**Issue by poetaster** [camelot-dev/camelot#470]
**Describe the bug**
This pdf, https://poetaster.de/misc/118.pdf (which I'm not uploading here since it may be a copyright issue) is read well but camelot shifts the rows under the multi-header controllability, down.  

**Steps to reproduce the bug**

Load the above file and try both stream and lattice reading.   I tried a lot of variations:

stream with different row tolerances:
```dfs = camelot.read_pdf('118.pdf', flavor='stream', row_tol=20,flag_size=True) ```

and lattice with many scale and shift variations.

```dfs = camelot.read_pdf('118.pdf', flavor='lattice', shift_text=['r','t', 'r', 't'], line_scale=20) ```

![dataframe](https://github.com/camelot-dev/camelot/assets/1065933/8a531cef-a835-42b9-bc4f-b21e1cae271a)

Lattice appears to get it right:

```camelot.plot(dfs[0], kind='grid').show()```

![lattice](https://github.com/camelot-dev/camelot/assets/1065933/ec24ad8f-e70a-4182-85e9-9f51b511209c)

Which seems correct.  But it always shifts the rows in the controllability part.

**Expected behavior**

Rows should not be shifted.

**Code**

a

Began with:

```
import camelot
dfs = camelot.read_pdf('118.pdf') 

```

And tried many variation, most recent lattice being:
```dfs = camelot.read_pdf('118.pdf', flavor='lattice', shift_text=['r','t', 'r', 't'], line_scale=20) ```

# add your code here
```

**PDF**
See above.

<!-- Add the PDF file that you want to extract tables from. -->

**Screenshots**
See above.

<!-- If applicable, add screenshots to help explain your problem. -->

**Environment**

- OS: ubuntu 22
- Python version: 3.10.12
- Numpy version: 1.24.0
- OpenCV version: 4.8.1.78
- Ghostscript version: 0.7
- Camelot version: 0.11.0

**Additional context**

<!-- Add any other context about the problem here. -->

**Comment by paulobarrera14** [comment]
@poetaster Pinging this again, where you able to find a fix for the multi-row headers?

**Comment by poetaster** [comment]
I hadn't worked on this (ended up reading excel files directly for that  project) since then. I've looked now, but thought i should probably update camelot? What version would be best to test with?

**Comment by poetaster** [comment]
I wasn't sure if I had done the original on my PC or on my jupyter lab server. On this pc, camelot is at 0.9.0 and the results are the same.
![camelot](https://github.com/camelot-dev/camelot/assets/1065933/9fc2bc9d-e442-4e62-9b26-e1b8aa907257)

**Comment by poetaster** [comment]
Ok, updated to 0.11.0 and same same. I'm not sure if it's just that I haven't understood the shifting 'foo', but even without, camelot get's the grid correct, but shifts the content of the 'controlability' columns 2 down.

**Comment by bosd** [comment]
Triage notes:

The shifted-rows symptom in your screenshots maps to a known interaction between multi-row column headers and the lattice grid: the textlines for the body rows have y-coordinates that fall just inside the top header row's grid cell when their visual cell is taller than a single row (because the header is two rows merged). Camelot's `get_table_index` assigns each textline to the row whose midpoint contains the textline's y-midpoint — when the merged header pushes that midpoint into the adjacent cell, body rows get shifted relative to the visual layout.

Two things worth re-testing on the 2.0 prerelease (the repo's master is now substantially different from the 0.11.0 you tested with):

1. **`flavor='hybrid'`** (didn't exist in 0.11.0). It combines the lattice grid detection with text-edge based flow, which often recovers the right row alignment in cases where the lattice grid is correct but the row-assignment step doesn't.

2. **`copy_text=['v']`** — your screenshot shows the lattice grid is correctly detected, so vertical-span copying may already pull the right values into the right rows. (Note: the [#349 fix](https://github.com/camelot-dev/camelot/pull/757) just landed and corrects a 2D-span loss in this exact code path — worth pulling master.)

If those don't help, the actual algorithmic fix is to change the row-assignment heuristic from 'midpoint-of-textline falls in row range' to 'textline's bounding-box-top is closest to row's top' (so cells whose top y-coordinate matches the visual row top get the textline regardless of merged-cell midpoint drift). That's a more invasive change — happy to attempt it as a follow-up PR if you can confirm whether `flavor='hybrid'` and/or the master copy_text fix still leave you stuck. Re-run with:

```bash
pip install -U git+https://github.com/camelot-dev/camelot.git
```

against 118.pdf and post the new screenshots.

**Comment by bosd** [comment]
**Resolved by #385 (PR #773).**

#470 is the same bug family as #385: cell text was accumulating in *processing order* (`Cell.text` is an appending setter, and `compute_parse_errors` processed all vertical-orientation textlines before horizontal ones), which displaced rows under multi-row headers.

PR #773 makes `compute_parse_errors` assemble cell text in global reading order `(-y0, x0)` across both orientations. Re-running `118.pdf` (`flavor='lattice'`, `line_scale=40`) with that fix applied, the controllability rows now line up correctly:

```
S2 E1 -> QM QM A
S2 E2 -> QM  A B
S3 E1 -> QM  A B
...
```

Every (severity, exposure) row carries its three controllability values on the correct row — the "rows shifted down" symptom is gone.

Two minor cosmetic residuals remain, neither being the row-shift: a footnote marker `a` attached to a `C` cell, and the multi-line A/B/C severity legend rendered as one cell — both reasonable extractions of what's in the PDF.

Will close once #773 lands on master. Tracked together with #385.

**Comment by bosd** [comment]
Fixed by #385 (PR #773, now merged). Verified against 118.pdf: the controllability rows under the multi-row header line up correctly. The two residuals (a footnote 'a' on a 'C' cell, the A/B/C legend as one multi-line cell) are correct extractions, not row displacement.
